python_onnxifi/data_conversion.cc

```diff
@@ -0,0 +1,298 @@
+#include "python_onnxifi/data_conversion.h"
+
+#include <algorithm>
+#include <complex>
+#include <functional>
+namespace py = pybind11;
+
+DescriptorData::DescriptorData() {}
+
+DescriptorData::DescriptorData(DescriptorData&& d) noexcept {
+  name = std::move(d.name);
+  buffer = std::move(d.buffer);
+  shape = std::move(d.shape);
+  descriptor.name = name.c_str();
+  descriptor.dimensions = shape.size();
+  descriptor.shape = shape.data();
+  descriptor.buffer = reinterpret_cast<onnxPointer>(buffer.data());
+  descriptor.memoryType = d.descriptor.memoryType;
+  descriptor.dataType = d.descriptor.dataType;
+}
+
+DescriptorData::DescriptorData(const DescriptorData& d) {
```

Comment:
The descriptor data can't be allocated on the stack, so keep it persisting i've put it as member variables. Since associated with each tensor descriptor are shape, name, and buffer pointers I group them in a struct. To keep the descriptor values correct, I've modified to copy and move constructors as well.

Reply:
> The descriptor data can't be allocated on the stack

What does that mean? Which data are we trying to allocate on stack?

Reply:
The memory for storing the data, the shape array, and the name need to be allocated when we make the tensor descriptors from numpy array. It wouldn't work to <std::vector<uint32_t> shapes>

Reply:
OK. I remember now.